packages/lesswrong/components/sequences/TopPostsPage.tsx

```diff
@@ -0,0 +1,906 @@
+import { useLocation } from '../../lib/routeUtil';
+import { Components, fragmentTextForQuery, registerComponent } from '../../lib/vulcan-lib';
+import { postGetPageUrl } from '../../lib/collections/posts/helpers';
+import { Link } from '../../lib/reactRouterWrapper';
+import { preferredHeadingCase } from '../../themes/forumTheme';
+import { useDisplayedPost, usePostBySlug } from '../posts/usePost';
+import { LWReviewWinnerSortOrder, getCurrentTopPostDisplaySettings } from './TopPostsDisplaySettings';
+
+import { gql, useQuery } from '@apollo/client';
+import classNames from 'classnames';
+import range from 'lodash/range';
+import { ReviewSectionInfo, ReviewWinnerSectionName, ReviewWinnerYear, reviewWinnerSectionsInfo, reviewWinnerYearGroupsInfo } from '../../lib/publicSettings';
+
+/** In theory, we can get back posts which don't have review winner info, but given we're explicitly querying for review winners... */
+type GetAllReviewWinnersQueryResult = (PostsTopItemInfo & { reviewWinner: Exclude<PostsTopItemInfo['reviewWinner'], null> })[]
+
+type ExpansionState = 'expanded' | 'collapsed' | 'default';
+type HiddenState = 'full' | 'hidden';
+export type CoordinatePosition = 'left' | 'middle' | 'right';
+
+interface PostGridDimensions {
+  postGridColumns: number;
+  postGridRows: number;
+}
+
+interface GetPostsInGridArgs extends PostGridDimensions {
+  posts: PostsTopItemInfo[];
+  viewportWidth: number;
+  viewportHeight: number;
+  leftBookOffset: number;
+}
+
+interface PostGridContentsProps extends PostGridDimensions {
+  postsInGrid: (PostsTopItemInfo | null)[][];
+  viewportHeight: number;
+  classes: ClassesType<typeof styles>;
+  id: string;
+  handleToggleFullyOpen: (id: string) => void;
+  gridContainerHeight: number;
+  isExpanded: boolean;
+  isShowingAll: boolean;
+  leftBookOffset: number;
+  coverLoaded: boolean;
+}
+
+interface PostGridCellContentsProps extends Omit<PostGridContentsProps, 'postsInGrid'> {
+  post: PostsTopItemInfo | null;
+  rowIdx: number;
+  columnIdx: number;
+}
+
+interface GetSplashArtUrlArgs {
+  reviewWinnerArt: ReviewWinnerTopPostsPage_reviewWinnerArt;
+  leftBookOffset: number;
+}
+
+const MAX_GRID_SIZE = 6;
+
+// TODO: not actually sure we want to use default coordinates if coordinates are missing
+// And even if we do, these values are definitely wrong (i.e. need to figure out aspect ratio)
+const DEFAULT_SPLASH_ART_COORDINATES: Omit<SplashArtCoordinates, '_id' | 'reviewWinnerArtId'> = {
+  leftHeightPct: .2, leftWidthPct: .2, leftXPct: .2, leftYPct: .2, leftFlipped: false,
+  middleHeightPct: .2, middleWidthPct: .2, middleXPct: .2, middleYPct: .2, middleFlipped: false,
+  rightHeightPct: .2, rightWidthPct: .2, rightXPct: .2, rightYPct: .2, rightFlipped: false,
+};
+
+// TODO: update the description to be appropriate for this page
+const description = `All of ${siteNameWithArticleSetting.get()}'s posts, filtered and sorted however you want`;
```

Comment:
I think this is used for SEO + search result purposes.  Need to figure out what goes here.  @rjmk @Discordius @darkruby501